cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillLauncher.java

```diff
@@ -124,8 +125,11 @@ public void run(String[] args) throws Exception {
           runnable.destroy();
         }
       } finally {
-        // Delete the pod itself to avoid pod goes into CrashLoopBackoff
-        deletePod(podInfo);
+        // Delete the pod when pod deletion is enabled. When k8s job is submitted, pod deletion is disabled.
+        if (Arrays.stream(args).noneMatch(str -> str.equalsIgnoreCase(KubeMasterEnvironment.DISABLE_POD_DELETION))) {
+          // Delete the pod itself to avoid pod goes into CrashLoopBackoff
```

Comment:
it would be helpful to add more context to this comment while we still remember why it is being done. Was it was something about preview?